simulation_server.py

```diff
@@ -438,13 +452,6 @@ def kill_webots(self):
                     self.webots_process.kill()
                 self.webots_process = None
 
-            # remove unused _webots images
-            available_images = os.popen(
-                "docker images --filter=reference='*_webots:*' --format '{{.Repository}}'").read().split('\n')
-            running_images = os.popen("docker ps --format '{{.Image}}'").read().split('\n')
-            unused_images = ' '.join([i for i in available_images if i not in running_images])
-            if unused_images:
-                os.system(f"docker image rm {unused_images}")
             # remove dangling images, stopped containers, build cache, volumes and networks
             os.system("docker system prune --volumes -f")
```

Comment:
I would add a --filter="until=24h" or so to limit the re-download of the webots docker images while preserving disk space.
See https://github.com/cyberbotics/competition-record-action/blob/0c24817e12577e52875562ff8b5963bc211aa60e/metascript/competition.py#L125-L127

Reply:
Good idea, I wonder if we should add a whitelist functionality to prevent important images from being removed; on our server we have the theia-test images from benjamin, the opendr-on-webots-cloud from stefania and cyberbotics' webots images that might get deleted that way.

Perhaps we could have a file that contains a whitelist and in the script we add filter arguments to the prune function based on it?

Reply:
Yes, that seems to be a very good idea.
The whitelist could go into the json configuration file.